hpobench/benchmarks/ml_mmfb/tabular_benchmark.py

```diff
@@ -0,0 +1,230 @@
+        raise NotImplementedError
+
+    # pylint: disable=arguments-differ
+    def get_fidelity_space(self, seed: Union[int, None] = None) -> CS.ConfigurationSpace:
+        raise NotImplementedError
+
+    # pylint: disable=arguments-differ
+    def get_meta_information(self) -> Dict:
+        """ Returns the meta information for the benchmark """
+        return {'name': 'BaseTabularBenchmark',
+                'references': [],
+                'task_id': self.task_id,
+                'model': self.model
+                }
+
+    def _preprocess_configspace(self, config_space: CS.ConfigurationSpace) -> CS.ConfigurationSpace:
+        """ Converts floats to np.float32 """
+        for hp in config_space.get_hyperparameters():
+            hp.sequence = tuple(np.array(hp.sequence).astype(np.float32))
+            hp.default_value = np.float32(hp.default_value)
+        return config_space
+
+    def _total_number_of_configurations(self, space: str = "hyperparameters") -> int:
+        """ Returns the number of unique configurations in the parameter/fidelity space
+        """
+        count = 1
+        cs = self.configuration_space if space == "hyperparameters" else self.fidelity_space
+        for hp in cs.get_hyperparameters():
+            count *= len(hp.sequence)
+        return count
+
+    def _seeds_used(self) -> List:
+        return self.table.seed.unique().tolist()
+
+    def sample_hyperparamer(self, n: int = 1) -> Union[CS.Configuration, List]:
+        return self.configuration_space.sample_configuration(n)
+
+    def sample_fidelity(self, n: int = 1) -> Union[CS.Configuration, List]:
+        return self.fidelity_space.sample_configuration(n)
+
+    def get_global_min(self, metric: str = "acc"):
+        """ Retrieves the minimum (1 - metric) for train, validation and test splits
+        """
+        assert metric in self.global_minimums.keys(), \
+            "Not a valid metric: {}".format(list(self.global_minimums.keys()))
+        return self.global_minimums[metric]
+
+    def get_max_fidelity(self) -> Dict:
+        max_fidelity = dict()
+        for hp in self.fidelity_space.get_hyperparameters():
+            max_fidelity[hp.name] = np.sort(hp.sequence)[-1]
+        return max_fidelity
+
+    def get_fidelity_range(self) -> List:
+        fidelities = []
+        for hp in self.fidelity_space.get_hyperparameters():
+            if not isinstance(hp, CS.Constant) and len(hp.sequence) > 1:
+                fidelities.append((hp.name, hp.sequence[0], hp.sequence[-1]))
+        return fidelities
+
+    def _search_dataframe(self, row_dict, df):
+        query_stmt = self._build_query(row_dict)
+        result = df.query(query_stmt)
+        # TODO: What happens in this case? The objective function raises a TypeError.
+        if len(result) == 0:
+            return None
+        return result.iloc[0].loc['result']
+
+        # TODO: This created an out-of-bounds error. The idx mask should have been 2d, but was 1d.
```

Comment:
I will try and do some local testing, and some benchmarking of speed
it is surprising that it fails
it would be great if you can help me point to the failing case for me to debug